src/lib/token_list_gen.ts

```diff
@@ -345,3 +351,56 @@ export const arbListtoEtherscanList = (
   return list;
 };
 
+export const permitTest = async (pathOrUrl: string) => {
+  const l1TokenList = await getTokenListObj(pathOrUrl);
+  removeInvalidTokensFromList(l1TokenList)
+
+  const newList = await generateTokenList(l1TokenList, undefined, {
+    includeUnbridgedL1Tokens: false,
+  });
+  const etherscanData = arbListtoEtherscanList(newList);
+  let dict: { [key: string]: any } = {};
+  
+  const { l1 } = await getNetworkConfig();
+  const wallet = ethers.Wallet.createRandom().connect(l1.provider);
+  const spender = ethers.Wallet.createRandom().connect(l1.provider);
+  const value = ethers.utils.parseUnits("1.0", 18);
+  const deadline = ethers.constants.MaxUint256;
+
+  for(let i=0; i<newList.tokens.length; i++){
+    const tokenContract = new ethers.Contract(etherscanData[i].l1Address!, permitTokenAbi['abi'], wallet);
+
+    try {
+      const signature = await getCorrectPermitSig(wallet, tokenContract, spender.address, value, deadline);
```

Comment:
we can reduce the number of network requests here significantly by using multicall
if we try to parse a token list with like 1k entries this might take a bit too long

I guess we'd want to do something similar to https://github.com/OffchainLabs/arb-token-lists/pull/25